src/core/test/system/borrower/balance.t.sol

```diff
@@ -0,0 +1,132 @@
+// Copyright (C) 2019 Centrifuge
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.5.12;
+
+import "../system.sol";
+import "../users/borrower.sol";
+
+contract BalanceTest is SystemTest {
+
+    Borrower borrower;
+    address borrower_;
+        
+    function setUp() public {
+        baseSetup();
+        // setup users
+        borrower = new Borrower(address(shelf), address(distributor), currency_, address(pile));
+        borrower_ = address(borrower);
+    }
+    
+    function balanceTake() public {
+        uint initialShelfBalance = currency.balanceOf(address(shelf));
+        uint takeAmount = currency.balanceOf(address(junior));
+        borrower.balance();
+        assertPostConditionTake(takeAmount, initialShelfBalance);
+    }
+
+    function balanceGive() public {
+        uint initialJuniorBalance = currency.balanceOf(address(junior));
+        uint giveAmount = currency.balanceOf(address(shelf));
+
+        borrower.balance();
+        assertPostConditionGive(giveAmount, initialJuniorBalance);
+    }
+
+    function assertPreConditionTake(uint takeAmount) public {
+        // assert: borrowFromTranches is active
+        assert(distributor.borrowFromTranches());
+        // assert: tranche reserve has enough funds
+        assert(currency.balanceOf(address(junior)) == takeAmount);
+    }
+
+    function assertPostConditionTake(uint takeAmount, uint initialShelfBalance) public {
+        // assert: takeAmount > 0
+        assert(takeAmount > 0);
+        // assert: all funds transferred from tranche reserve
+        assertEq(currency.balanceOf(address(junior)), 0);
```

Comment:
Not in every case but If you only test cases where it is true. 

It should be okay

Reply:
in the switchable it is always the case :)